core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -354,10 +354,12 @@ class AdminManager(val config: KafkaConfig,
       }
       def createResponseConfig(configs: Map[String, Any],
                                createConfigEntry: (String, Any) => DescribeConfigsResponseData.DescribeConfigsResourceResult): DescribeConfigsResponseData.DescribeConfigsResult = {
-        val filteredConfigPairs = configs.filter { case (configName, _) =>
-          /* Always returns true if configNames is None */
-          resource.configurationKeys.asScala.forall(_.contains(configName))
-        }.toBuffer
+        val filteredConfigPairs = if (resource.configurationKeys == null)
```

Comment:
Will `Option(resource.configurationKeys).forall(_.asScala.forall(_.contains(configName)))` be better?

Reply:
Thanks @huxihx , I agree your suggestion makes the code less and simplified. But for better readability, I'd prefer the @tombentley 's suggestion. Thank you.